crates/astria-sequencer/src/transaction/mod.rs

```diff
@@ -42,6 +44,19 @@ pub(crate) async fn execute<S: StateWriteExt>(
         .await
 }
 
+#[derive(Debug)]
+pub(crate) struct InvalidNonce(pub(crate) u32);
```

Comment:
Add an `impl std::error::Error for InvalidNonce` here. No need to impl any of its methods, just use the default ones.

Reply:
use the derive macro thiserror, [here is an example ](https://github.com/ralexstokes/ethereum-consensus/blob/02a70ad7f9b8a1089011f6c27b77ef1854d86a2d/ethereum-consensus/src/state_transition/error.rs#L10-L57), or derive-more, [here is an example deriving `Display`](https://github.com/emhane/discv5/blob/4750490760931b3ba89afa44594f7187fd74a7b5/src/rpc/notification.rs#L16-L25) and [here is an example deriving `From`](https://github.com/emhane/discv5/blob/4750490760931b3ba89afa44594f7187fd74a7b5/src/error.rs#L6-L64) but `Error` is also available from that crate. disable default features and only bring the features into scope that you use, [like this](https://github.com/emhane/discv5/blob/4750490760931b3ba89afa44594f7187fd74a7b5/Cargo.toml#L41).

Reply:
in general, in rust, "Errors must describe themselves through the [Display](https://doc.rust-lang.org/std/fmt/trait.Display.html) and [Debug](https://doc.rust-lang.org/std/fmt/trait.Debug.html) traits. Error messages are typically concise lowercase sentences without trailing punctuation" (source:[ std library docs](https://doc.rust-lang.org/std/error/trait.Error.html))

Reply:
I agree about `thiserror`, but feels like overkill just for this. When we implement error handling across the sequencer we should probably do that though